### DISCUSSION TITLE
How To Restore?

### DISCUSSION THREAD

**Discussion by ccigas** [dani-garcia/vaultwarden#1468] [Q&A]
Hello everyone, I upgraded my server today and after I copied in all my files for BW, I assumed the vault and all the users would be all there in my new install. Turns out it didnt work the way I was thinking. Luckily I took a back up of my vault but completely forgot about my two family members using it.

I still have the copied off db files but I am not sure how to restore them. I stopped bitwarden in my docker compose yaml, copied the db over again and still no luck.

How can I restore with these db files?

**Comment by jjlin** [comment]
See https://github.com/dani-garcia/bitwarden_rs/wiki/Backing-up-your-vault

**Reply by ccigas** [reply]
Thanks but I mentioned I did that already.

Stopped my bw in docker compose, copied over the all the files from the previous install and started it back up with no luck. These files included the folder of icon_cache, config.json, dbsqlite3, db_20210228.sqlite3, db_20210228.sqlite3-shm, db_20210228.sqlite3-wal and rsa_key.pub

**Reply by jjlin** [reply]
Well, you actually didn't specify exactly what you did. The wiki page says you need to:

> replace each file or directory in the `data` dir with its backed up version

Did you actually do that? If you did, it should work. If it didn't work, your backup files are likely wrong or corrupted.

**Comment by stanHK** [comment]
Hi, I am having this issue too. 

1. I have installed Vaultwarden
2. Created a user
3. Created an organisation
4. Imported the passwords
5. Invited other users
6. Created some new collections

After the Docker went offline - power outage - items 1-4 were available. Item 5 and 6 not. 

As I run a copy of VW on another machine, I have tried to Zip the database and overwritten it on the other instance of VW. After logging in, exactly the same issue. 1-4 are available, not 5 and 6. 

So it seems impossible for me to backup the database including all users. Is this a known issue? Any workarounds? JSON of the passwords only imports organization, collections, and passwords, not users.

**Reply by ccigas** [reply]
> Hi, I am having this issue too. 
> 
> 
> 
> 1. I have installed Vaultwarden
> 
> 2. Created a user
> 
> 3. Created an organisation
> 
> 4. Imported the passwords
> 
> 5. Invited other users
> 
> 6. Created some new collections
> 
> 
> 
> After the Docker went offline - power outage - items 1-4 were available. Item 5 and 6 not. 
> 
> 
> 
> As I run a copy of VW on another machine, I have tried to Zip the database and overwritten it on the other instance of VW. After logging in, exactly the same issue. 1-4 are available, not 5 and 6. 
> 
> 
> 
> So it seems impossible for me to backup the database including all users. Is this a known issue? Any workarounds? JSON of the passwords only imports organization, collections, and passwords, not users. 
> 
> 
> 
> 

So the reason for my issue is that I created all of my containers not using sudo but on server restart, docker-compose starts with sudo. So not saying that could help you but that was my issue.
